src/Mod/Path/PathScripts/PathPost.py

```diff
@@ -41,18 +42,365 @@
 
 LOG_MODULE = PathLog.thisModule()
 
-PathLog.setLevel(PathLog.Level.INFO, LOG_MODULE)
+if False:
+    PathLog.setLevel(PathLog.Level.DEBUG, PathLog.thisModule())
+    PathLog.trackModule(PathLog.thisModule())
+else:
+    PathLog.setLevel(PathLog.Level.INFO, PathLog.thisModule())
 
 
 translate = FreeCAD.Qt.translate
 
+
 class _TempObject:
     Path = None
     Name = "Fixture"
     InList = []
     Label = "Fixture"
 
 
+def resolveFileName(job, subpartname, sequencenumber):
+    PathLog.track(subpartname, sequencenumber)
+
+    validPathSubstitutions = ["D", "d", "M", "j"]
+    validFilenameSubstitutions = ["j", "d", "T", "t", "W", "O", "S"]
+
+    # Look for preference default
+    outputpath, filename = os.path.split(PathPreferences.defaultOutputFile())
+    filename, ext = os.path.splitext(filename)
+
+    # Override with document default if it exists
+    if job.PostProcessorOutputFile:
+        matchstring = job.PostProcessorOutputFile
+        candidateOutputPath, candidateFilename = os.path.split(matchstring)
+
+        if candidateOutputPath:
+            outputpath = candidateOutputPath
+
+        if candidateFilename:
+            filename, ext = os.path.splitext(candidateFilename)
+
+    # Strip any invalid substitutions from the ouputpath
+    for match in re.findall("%(.)", outputpath):
+        if match not in validPathSubstitutions:
+            outputpath = outputpath.replace(f"%{match}", "")
+
+    # if nothing else, use current directory
+    if not outputpath:
+        outputpath = "."
+
+    # Strip any invalid substitutions from the filename
+    for match in re.findall("%(.)", filename):
+        if match not in validFilenameSubstitutions:
+            filename = filename.replace(f"%{match}", "")
+
+    # if no filename, use the active document label
+    if not filename:
+        filename = FreeCAD.ActiveDocument.Label
+
+    # if no extension, use something sensible
+    if not ext:
+        ext = ".nc"
+
+    # By now we should have a sanitized path, filename and extension to work with
+    PathLog.track(f"path: {outputpath} name: {filename} ext: {ext}")
```

Comment:
I got in the habit of using track() because the output included the line number.  debug() didn't.  Then I fixed debug() to add the line number and forgot to start using it.  doh!